src/WingetCreateTests/WingetCreateTests/UnitTests/UpdateCommandTests.cs

```diff
@@ -212,6 +212,62 @@ public async Task UpdateFailsWithUnmatchedPackages()
             Assert.That(result, Does.Contain(Resources.NewInstallerUrlMustMatchExisting_Message), "New installer must match error should be thrown");
         }
 
+        /// <summary>
+        /// Verify that update command warns if submit arguments are provided without submit flag being set.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Test]
+        public async Task UpdateChecksMissingSubmitFlag()
+        {
+            string packageId = "TestPublisher.TestPackageId";
+            string version = "1.2.3.4";
+
+            UpdateCommand command1 = new UpdateCommand
+            {
+                Id = packageId,
+                Version = version,
+                InstallerUrls = new[] { "https://fakedomain.com/fakeinstaller.exe" },
+                SubmitToGitHub = false,
+                Replace = true,
+            };
+
+            UpdateCommand command2 = new UpdateCommand
+            {
+                Id = packageId,
+                Version = version,
+                InstallerUrls = new[] { "https://fakedomain.com/fakeinstaller.exe" },
+                SubmitToGitHub = false,
+                PRTitle = "Test PR Title",
+            };
+
+            try
+            {
+                await command1.Execute();
+            }
+            catch (Exception)
+            {
+                // Expected exception
+            }
+
+            string result = this.sw.ToString();
+            Assert.That(result, Does.Contain(Resources.SubmitFlagMissing_Warning), "Submit arguments provided without submit flag warning should be thrown");
+
+            // Clear the string writer
+            this.sw.GetStringBuilder().Clear();
+
+            try
+            {
+                await command2.Execute();
+            }
+            catch (Exception)
+            {
+                // Expected exception
+            }
+
+            result = this.sw.ToString();
+            Assert.That(result, Does.Contain(Resources.SubmitFlagMissing_Warning), "Submit arguments provided without submit flag warning should be thrown");
+        }

```

Comment:
Nit: This can be broken up into two separate unit tests named accordingly. That way you don't have to clear the string writer.